java/server/src/org/openqa/grid/internal/BaseRemoteProxy.java

```diff
@@ -342,7 +309,7 @@ public TestSession getNewSession(Map<String, Object> requestedCapability) {
       return null;
     }
     // any slot left for the given app ?
-    for (TestSlot testslot : testSlots) {
+    for (TestSlot testslot : getTestSlots()) {
```

Comment:
It shouldn't be a problem because the only place wherein we would like to leverage the logic of `BaseRemoteProxy` is when we are iterating the `TestSlot` list for new session creation. But just to ensure that we are consistent across the board, I have fixed all the places to access the `testSlots` list only via the `getTestSlots()` method.